MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/ProductPageController.java

```diff
@@ -213,19 +198,124 @@ public String delProduct(@RequestParam int productId){
+        for(Store s : product.getStores()) {
+            storesInProductById.put(s.getId(),s.getName() + ", " + s.getAddress());
+            listStoresInProductById.add(s.getId());
+        }
+
+        Map<Integer,String> allStoresById = new HashMap<>();
+        for(Store s : allStores) {
+            allStoresById.put(s.getId(),s.getName() + ", " + s.getAddress());
+        }
+
+        StoresInProduct storesInProduct = new StoresInProduct();
+        storesInProduct.setStoresId(listStoresInProductById);
+
+        model.addAttribute("storesId", allStoresById);
+        model.addAttribute("storesInProduct", storesInProduct);
+        model.addAttribute("product", product);
+
+        return "productInStores";
 
     }
 
+    @RequestMapping(value = "/stores", method = RequestMethod.POST)
+    public String getStoresByProductId(@ModelAttribute("storesInProduct") StoresInProduct storesInProduct, @RequestParam int productId) {
+
+        final String getStoresUri = productUrl + "/{productId}/productStores/{userId}";
+        final String getStoreByIdUri = storeUrl + "/{storeId}";
+        final String getProductByIdUri = productUrl + "/{productId}";
+        final String addStoreToProductUri = productUrl + "/stores/";
+        final String deleteStoreFromProductUri = productUrl + "/deleteStores/";
+
+        int userId = getCurrentUser().getId();
+
+        Map<String, Integer> param = new HashMap<>();
+        param.put("productId", productId);
+        param.put("userId", userId);
+
+        RestTemplate restTemplate = getRestTemplate();
+
+        Store[] result = restTemplate.getForObject(getStoresUri, Store[].class, param);
+        List<Store> oldStores = Arrays.asList(result);
+
+        Product product = restTemplate.getForObject(getProductByIdUri, Product.class, param);
+
+        //Get list of stores named "newStores", that had been checked in the form
+
+        List<Store> newStores = new ArrayList<>();
+
+        for(int storeId : storesInProduct.getStoresId()) {
+            param.clear();
+            param.put("storeId", storeId);
+            newStores.add(restTemplate.getForObject(getStoreByIdUri, Store.class, param));
+        }
+
+        List<Store> storesToAdd = new ArrayList<>();
+        List<Store> storesToDelete = new ArrayList<>();
+
+        if(newStores != null) {
+            outer:for(Store ns : newStores) {
+                for(Store os : oldStores) {
+                    if(ns.getId() == os.getId()) {
+                        continue outer;
+                    }
+                }
+                storesToAdd.add(ns);
+            }
+
+            outer:for(Store os : oldStores) {
+                for(Store ns : newStores) {
+                    if(os.getId() == ns.getId()) {
+                        continue outer;
+                    }
+                }
+                storesToDelete.add(os);
+            }
+        } else {
+            storesToDelete.addAll(oldStores);
+        }
```

Comment:
How I would rewrite this loops:
```
List<Store> storesToAdd = new ArrayList<>(newStores);
List<Store> storesToDelete = new ArrayList<>(oldStores);

storesToAdd.removeAll(oldStores);
storesToDelete.removeAll(newStores);
```
Does it make sens? Is it easier to understand?

However, for this, you need to have proper equals method implemented in the Store class, or you can do this with IDs and then get real Store object, what can reflect on performance, less amount of data to send over the network.